### PR TITLE
Add discussion templates

### PR DESCRIPTION
**Pull request by SMillerDev**
CC @p-linnane

### CODE REVIEW

File: .github/DISCUSSION_TEMPLATE/casks.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+label: ["cask"]
+body:
+  - type: textarea
+    attributes:
+      label: Description of issue
+    validations:
+      required: true
+  - type: input
+    attributes:
+      label: Command that failed
```
Comment (p-linnane) [inline .github/DISCUSSION_TEMPLATE/casks.yml:11]:
```suggestion
      label: Output of command that failed with `--verbose`
      render: shell
```

Comment (p-linnane) [reply .github/DISCUSSION_TEMPLATE/casks.yml:11]: Not sure if we want to do this or not. Same with possibly making it required like the other fields.

Comment (SMillerDev) [reply .github/DISCUSSION_TEMPLATE/casks.yml:11]: I actually left it optional and non-verbose on purpose. We should just be outputting enough info in a normal run to get a basic understanding of the issue. And not all questions will be about commands.

File: .github/DISCUSSION_TEMPLATE/everyday-usage.yml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+label: ["simple"]
+body:
+  - type: textarea
+    attributes:
+      label: Description of issue
+    validations:
+      required: true
+  - type: input
+    attributes:
+      label: Command that failed
```
Comment (MikeMcQuaid) [inline .github/DISCUSSION_TEMPLATE/everyday-usage.yml:11]: Maybe can just omit this?

If not: let's make it more specific and make required things first

File: .github/DISCUSSION_TEMPLATE/getting-started.yml [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+label: ["simple", "getting-started"]
+body:
+  - type: textarea
+    attributes:
+      label: Description of issue
+    validations:
+      required: true
+  - type: input
+    attributes:
+      label: Command that failed
```
Comment (MikeMcQuaid) [inline .github/DISCUSSION_TEMPLATE/getting-started.yml:11]: Same here: remove it and make require things first.

File: .github/DISCUSSION_TEMPLATE/linux.yml [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+        - label: I am using Windows Subsystem for Linux
+          required: false
+  - type: textarea
+    attributes:
+      label: Description of issue
+    validations:
+      required: true
+  - type: input
+    attributes:
+      label: Command that failed
```
Comment (MikeMcQuaid) [inline .github/DISCUSSION_TEMPLATE/linux.yml:17]: Same here: remove it and make require things first.

File: .github/DISCUSSION_TEMPLATE/linux.yml [hunk 5]
```diff
@@ -0,0 +1,31 @@
+name: Linux
+label: ["linux"]
+body:
+  - type: checkboxes
+    attributes:
+      label: Additional Info
+      options:
+        - label: I am using Windows Subsystem for Linux
```
Comment (MikeMcQuaid) [inline .github/DISCUSSION_TEMPLATE/linux.yml:8]: Do we want info on other distros?

File: .github/DISCUSSION_TEMPLATE/writing-formulae-casks.yml [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+label: ["advanced"]
+body:
+  - type: textarea
+    attributes:
+      label: Description of issue
+    validations:
+      required: true
+  - type: input
+    attributes:
+      label: Command that failed
```
Comment (MikeMcQuaid) [inline .github/DISCUSSION_TEMPLATE/writing-formulae-casks.yml:11]: Same here: remove it and make require things first.

File: .github/DISCUSSION_TEMPLATE/writing-formulae-casks.yml [hunk 7]
```diff
@@ -0,0 +1,25 @@
+name: Writing formulae and casks
```
Comment (MikeMcQuaid) [inline .github/DISCUSSION_TEMPLATE/writing-formulae-casks.yml:1]:
```suggestion
name: Developing formulae or casks
```

maybe?

File: .github/DISCUSSION_TEMPLATE/tap-maintenance-and-brew-development.yml [hunk 8]
```diff
@@ -0,0 +1,25 @@
+name: Tap maintenance and Brew development
```
Comment (MikeMcQuaid) [inline .github/DISCUSSION_TEMPLATE/tap-maintenance-and-brew-development.yml:1]:
```suggestion
name: Tap maintenance and Homebrew development
```

File: .github/DISCUSSION_TEMPLATE/tap-maintenance-and-brew-development.yml [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+label: ["advanced"]
+body:
+  - type: textarea
+    attributes:
+      label: Description of issue
+    validations:
+      required: true
+  - type: input
+    attributes:
+      label: Command that failed
```
Comment (MikeMcQuaid) [inline .github/DISCUSSION_TEMPLATE/tap-maintenance-and-brew-development.yml:11]: Same here: remove it and make require things first.

File: .github/DISCUSSION_TEMPLATE/linux.yml (hunk 5 above)
Comment (SMillerDev) [reply .github/DISCUSSION_TEMPLATE/linux.yml:8]: I don't think it's needed. Brew doctor will reveal that.

Comment (MikeMcQuaid) [reply .github/DISCUSSION_TEMPLATE/linux.yml:8]: Will `brew doctor` or `brew config` reveal WSL?

Comment (p-linnane) [reply .github/DISCUSSION_TEMPLATE/linux.yml:8]: We are covered by `brew config`. WSL kernels show up like `Linux 5.15.57.1-Microsoft`.

Comment (SMillerDev) [reply .github/DISCUSSION_TEMPLATE/linux.yml:8]: I think this is easier to parse for maintainers, and giving more issues than just different distros.

Comment (p-linnane) [reply .github/DISCUSSION_TEMPLATE/linux.yml:8]: That makes sense to me. I think this is ready to merge and iterate on based off feedback.

**Comment by SMillerDev** [comment]
Well, that would have been great to know earlier.
Screenshot 2022-12-29 at 17 02 59

**Pull request merged**